### PR TITLE
Explicitly launch code server on 0.0.0.0

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

After vscode server upstream changes, the server now launches on `127.0.0.1` by default, so I updated the startup script to launch explicitly on `0.0.0.0`

## How to test

1. Select insiders 
2. Launch a workspace
3. Run on terminal `sudo lsof -i -P -n | grep 23000 | grep LISTEN` or `netstat -anlp | grep 23000 | grep LISTEN` and check bound address is `0.0.0.0`

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jeanp413** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by filiptronicek — APPROVED** [review]
Looks good to me ✅ ❤️ 


Attached is my `netstat` output:

```
gitpod /workspace/template-typescript-node $ netstat -anlp | grep 23000 | grep LISTEN
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 0.0.0.0:23000           0.0.0.0:*  
```

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 76e72e748ed6100e8be208a568c27372b7d1e292</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7616#pullrequestreview-853335412" title="Approved">filiptronicek</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [filiptronicek]
- ~~[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)~~ [filiptronicek]
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [filiptronicek]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7616#issuecomment-1013438461" title="Approved">filiptronicek</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7616#issuecomment-1013438461" title="Approved">filiptronicek</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [filiptronicek]
- ~~[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)~~ [filiptronicek]
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [filiptronicek]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/supervisor/pkg/supervisor/supervisor.go
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/supervisor.go:816]: @filiptronicek @jeanp413 You tested that it does not affect stable version?

Comment (jeanp413) [reply components/supervisor/pkg/supervisor/supervisor.go:816]: @akosyakov yeah it's fine as it's not a valid argument

**Pull request merged**
